src/nunit.analyzers/TestMethodAccessibilityLevel/TestMethodAccessibilityLevelAnalyzer.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.Extensions;
+
+namespace NUnit.Analyzers.TestMethodAccessibilityLevel
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class TestMethodAccessibilityLevelAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor testMethodIsNotPublic = DiagnosticDescriptorCreator.Create(
+            id: AnalyzerIdentifiers.TestMethodIsNotPublic,
+            title: TestMethodAccessibilityLevelConstants.TestMethodIsNotPublicTitle,
+            messageFormat: TestMethodAccessibilityLevelConstants.TestMethodIsNotPublicMessage,
+            category: Categories.Structure,
+            defaultSeverity: DiagnosticSeverity.Error,
+            description: TestMethodAccessibilityLevelConstants.TestMethodIsNotPublicDescription);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(testMethodIsNotPublic);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxNodeAction(AnalyzeMethod, SyntaxKind.MethodDeclaration);
+        }
+
+        private static void AnalyzeMethod(SyntaxNodeAnalysisContext context)
+        {
+            var methodNode = context.Node as MethodDeclarationSyntax;
+
+            if (methodNode != null && !methodNode.ContainsDiagnostics)
+            {
+                var testCaseType = context.SemanticModel.Compilation.GetTypeByMetadataName(NunitFrameworkConstants.FullNameOfTypeTestCaseAttribute);
+                var testType = context.SemanticModel.Compilation.GetTypeByMetadataName(NunitFrameworkConstants.FullNameOfTypeTestAttribute);
```

Comment:
why do you need those two variables? Seems like they are not used?

Reply:
Initially, I kept them to ensure that we are in a NUnit context. But `IsTestMethod` serves the same purpose, so you are right that these are not needed.